### PR TITLE
Validate if peer ID is empty

### PR DESCRIPTION
**Pull request by kishansagathiya**
Fixes https://github.com/libp2p/go-libp2p/issues/370

Let me know if tests for this change are desirable.

### CODE REVIEW

File: peer.go [hunk 1]
```diff
@@ -92,6 +94,15 @@ func (id ID) ExtractPublicKey() (ic.PubKey, error) {
 	return pk, nil
 }
 
+// Validate check if ID is empty or not
+func (id ID) Validate() error {
+	if id == ID("") {
+		return errors.New("Empty peer ID")
```
Comment (Stebalien) [inline peer.go]: This should probably:

1. Be a global `ErrEmptyPeerID`.
2. Be "empty peer ID" (lower case, see: https://github.com/golang/go/wiki/CodeReviewComments#error-strings)

**Review by Stebalien — CHANGES_REQUESTED** [review]
Thanks!  A couple of small nits and this should be good to go.

Eventually, we'll also want to validate that the peer ID is a valid multihash but I don't think we can do that efficiently at the moment (so we'll have to do it later).

**Comment by Stebalien** [comment]
This should also have a simple test case.

File: peer.go [hunk 2]
```diff
@@ -21,6 +21,22 @@ import (
...
+func NewErrEmptyPeerID() *ErrEmptyPeerID {
+	return &ErrEmptyPeerID{
+		message: "empty peer ID",
+	}
+}
+
+func (e *ErrEmptyPeerID) Error() string {
+	return e.message
+}
+
```
Comment (kishansagathiya) [inline peer.go]: did instead you ask for something like
```
const ErrEmptyPeerID = "empty peer ID"
```
?

Current approach is inspired from https://medium.com/@sebdah/go-best-practices-error-handling-2d15e1f0c5ee

File: peer_test.go [hunk 3]
```diff
@@ -204,6 +204,27 @@ func TestPublicKeyExtraction(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = p.Validate()
+	if err != nil {
+		t.Error("expected nil, but found " + err.Error())
+	}
+}
+
```
Comment (kishansagathiya) [inline peer_test.go:227]: I Have used t.Error() for function that is being tested here. For failure of any other functions t.Fatal() has been used. Kinda inspired from other tests here,

File: peer.go (hunk 2 above)
Comment (kishansagathiya) [reply peer.go]: Just saw this 

https://github.com/libp2p/go-libp2p-swarm/blob/a00fa8aa37a812f404bfa9b9d3dccc82a66029cf/swarm_dial.go#L30-L44

Let me add changes

**Review by Stebalien — APPROVED** [review]

**Pull request merged**